feature-libs/cart/import-export/core/services/import-csv.service.ts

```diff
@@ -49,12 +36,59 @@ export class ImportCsvService {
   }
 
   loadCsvData(file: File): Observable<string[][]> {
-    return this.loadFile(file).pipe(
-      map((res) => this.readCsvData(res as string))
-    );
+    return this.importService
+      .loadTextFile(file)
+      .pipe(map((res) => this.readCsvData(res as string)));
+  }
+
+  isReadableFile(
+    isDataParsable?: (data: string[][]) => boolean
+  ): AsyncValidatorFn {
```

Comment:
This service could be totally agnostic to the feature cart-import-export, so it can be potentially reused in other CSV-related features in the future. It suffices this method would be parametrized with the validation options:
```
{
   separator?: string,
   isDataParsable?: (data: string[][]) => boolean,
   maxEntries?: number
}
```

Reply:
yup, already done in pararell PR https://github.com/SAP/spartacus/pull/13895